build_functions/CMakeLists.txt

```diff
@@ -109,6 +112,67 @@ function(build_lib whatIsBuilding solution_folder)
         target_sources(${whatIsBuilding}_lib_${CMAKE_PROJECT_NAME} PRIVATE ${MOCKS_SOURCE_FILE})
     endif()
 
+    # ENABLE_TEST_FILES_PRECOMPILED_HEADERS creates precompiled headers for the test file
+    # It expects that exactly one test file exists. This is because cmake's target_precompile_headers
+    # only allows one precompiled headers file.
+    # If a specific header filename is provided, that is used.
+    # If just the flag is provided, the header filename is by convention computed from the test file by replacing the extension with .h
+    # A _mocks.c file is generated that includes the precompiled header without ENABLE_MOCKS_DECL,
+    # having the purpose of providing the mock implementations.
+    if(arg_ENABLE_TEST_FILES_PRECOMPILED_HEADERS)
+        # Check that exactly one test file exists (required for all cases)
+        list(LENGTH ${whatIsBuilding}_test_files TEST_FILES_COUNT)
+        if(NOT TEST_FILES_COUNT EQUAL 1)
+            message(WARNING "ENABLE_TEST_FILES_PRECOMPILED_HEADERS requires exactly one test file, but found ${TEST_FILES_COUNT} files in ${whatIsBuilding}_test_files: ${${whatIsBuilding}_test_files}")
```

Comment:
>WARNING [](http://example.com/codeflow?start=20&length=7)

nice logging, but should then be FATAL, because the project cannot be generated.